src/App.jsx

```diff
@@ -133,8 +161,23 @@ function App() {
       const todo = {
         id: addedTodo.id,
         title: addedTodo.fields.title,
+        createdDate: addedTodo.createdTime,
       };
-      setTodoList([...todoList, todo]);
+      setTodoList((prevTodoList) => {
+        const updatedList = [...prevTodoList, todo].sort((a, b) => {
+          if (sortField === "title") {
```

Comment:
This sorting logic is duplicated with the logic in fetchData! I think we should do one of two things:

1. Pull the sorting code into its own helper function (so you only write it once), or, even better...


2. Remove the useCallback from the fetchData function (we only should re-fetch all the data when we have to, which IMO is when we refresh the page). Instead, let's add a `useCallback` that takes the `todoList`, `sortField`, and `sortOrder` as dependencies and just sorts `todoList` according to the current field/order.

We don't need to re-fetch any data when we change our sort ordering. We already have all the data here! Putting only the sorting in the `useCallback` prevents the specific expensive code from running every re-run. We instead only sort if the data has changed, or our sorting criteria has changed.

With that, you can change your useEffect that calls `fetchData` back to an empty dependency array so that it runs only at page loadup:
```
useEffect(() => {...}, [])
```